api/src/handlers.rs

```diff
@@ -1,5 +1,237 @@
+            connection_string: account_connection,
+            database_name: "CribbageDb".to_owned(),
+            collection_name: "registeredAis".to_owned(),
+            account_name: "rust-cribbage-db".to_owned(),
+        };
+
+        cci
+    }};
+}
+#[allow(unused_macros)]
+macro_rules! cosmos_connection {
+    ($connection_info: expr) => {{
+        let authorization_token = match AuthorizationToken::new_master(&$connection_info.key) {
+            Ok(authorization_token) => authorization_token,
+            Err(e) => {
+                return HttpResponse::Ok().body(e.to_string());
+            }
+        };
+
+        let client = match ClientBuilder::new(&$connection_info.account_name, authorization_token) {
+            Ok(client) => client,
+            Err(e) => {
+                return HttpResponse::Ok().body(e.to_string());
+            }
+        };
+
+        client
+    }};
+}
+
+pub async fn get_registered_ais() -> impl Responder {
+    // let mut ai = CribbageAi {
+    //     name: "Hard".to_owned(),
+    //     by: "joe".to_owned(),
+    //     description: "used drop table to optimize crib".to_owned(),
+    //     uri: "localhost:8081".to_owned(),
+    // };
+
+    // let mut lst: Vec<CribbageAi> = Vec::<CribbageAi>::new();
+    // lst.push(ai);
+    // ai = CribbageAi {
+    //     name: "Easy".to_owned(),
+    //     by: "joe".to_owned(),
+    //     description: "ignores crib cards and optimizes for hand only".to_owned(),
+    //     uri: "localhost:8082".to_owned(),
+    // };
+
+    // lst.push(ai);
+
+    // ai = CribbageAi {
+    //     name: "Random".to_owned(),
+    //     by: "joe".to_owned(),
+    //     description: "picks a valid but random card to play".to_owned(),
+    //     uri: "localhost:8083".to_owned(),
+    // };
+    // lst.push(ai);
+    // let json = serde_json::to_string(&lst).unwrap();
+
+    let connection_info = get_connection_info!();
+    let client = cosmos_connection!(connection_info);
+    let client = client.with_database_client(&connection_info.database_name);
+    let client = client.with_collection_client(connection_info.collection_name);
+
+    let response = match client
+        .list_documents()
+        .with_max_item_count(100)
+        .execute::<CribbageAi>()
+        .await
+    {
+        Ok(response) => response,
+        Err(e) => {
+            return HttpResponse::Ok().body(e.to_string());
+        }
+    };
+
+    let json = serde_json::to_string(&response.documents).unwrap();
+
+    HttpResponse::Ok().body(json)
+}
+
+pub async fn add_ai(info: Path<(String, String, String, String)>) -> impl Responder {
+    let info = info.into_inner();
+    let ai = CribbageAi {
+        name: info.0,
+        by: info.1,
+        description: info.2,
+        uri: info.3,
+    };
+
+    let connection_info = get_connection_info!();
+
+    let client = cosmos_connection!(connection_info);
+
+    let client = client.into_database_client(&connection_info.database_name);
```

Comment:
Can you chain these calls for readability, instead of the multiple `let client = client.` lines?